core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -813,41 +813,56 @@ class Partition(val topicPartition: TopicPartition,
    */
   private def tryCompleteDelayedRequests(): Unit = delayedOperations.checkAndCompleteAll()
 
+  // Visible for testing.
+  private[cluster] def shouldShrinkIsr(): Boolean = {
+    leaderLogIfLocal match {
```

Comment:
nit: this might be simpler since we don't actually need the `Log`?
```scala
    if (isLeader) {
      inReadLock(leaderIsrUpdateLock) {
        getOutOfSyncReplicas(replicaLagTimeMaxMs).nonEmpty
      }
    }
```

Reply:
Yes. Done.